src/api/cpp/array.cpp

```diff
@@ -20,6 +19,12 @@
 #include <af/internal.h>
 #include "error.hpp"
 
+#include <stdexcept>
+#include <memory>
+
+
+using std::unique_ptr;
```

Comment:
I originally implemented array_proxy_impl using unique ptr but it became more complicated than it needed to be. Removing.